spicy/toolchain/src/ast/types/unit.cc

```diff
@@ -38,6 +40,27 @@ hilti::optional_ref<const type::unit::Item> type::Unit::itemByName(const ID& id)
     return {};
 }
 
+std::optional<uint64_t> type::Unit::size() const {
+    uint64_t result = 0;
+    for ( const auto& i : this->itemRefs() ) {
+        if ( const auto& f = i->tryAs<unit::item::Field>() ) {
+            // TODO(bbannier): To support fields with conditions we need to be able to return an runtime value (an
+            // `Expression`) instead of a compile-time `uint64_t` here which requires support in the type API.
+            if ( f->condition() )
+                return {};
+
+            // TODO(bbannier): Add support for returning value of fixed-size lists. These will not be represented as
+            // vectors in the unit but as opaque counters, so this requires some more work elsewhere.
+            if ( auto s = f->parseType().size() )
```

Comment:
Couldn't this be `f->size()`?

Reply:
We are implementing `Type::size()` here which needs to return a static size; `Field::size()` OTOH hand returns an `Expression` (e.g.., to support `&size`).

What this means is that we can only really compute units which exclusively contain fields with attribute-independent sizes, i.e., only units over integers. This is not great, and in light of your comment below we maybe should remove this code here and only truely skip fields of unit type if they come with e.g., a `&size`.

Reply:
I now removed this code again.